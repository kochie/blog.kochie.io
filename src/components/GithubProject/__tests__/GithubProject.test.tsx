import React from 'react'
import { ReactTestRenderer, act, create } from 'react-test-renderer'
import { jest } from '@jest/globals'
import * as Sentry from '@sentry/nextjs'
import sentryTestkit from 'sentry-testkit'

const { sentryTransport } = sentryTestkit()

const SENTRY_DSN = process.env.SENTRY_DSN || process.env.NEXT_PUBLIC_SENTRY_DSN
// initialize your Sentry instance with sentryTransport
Sentry.init({
  dsn:
    SENTRY_DSN ||
    'https://93d1763e43d24b2885baed0a99a74b02@o157203.ingest.sentry.io/5779239',
  transport: sentryTransport,
  //... other configurations
})

type Path = [string, { [key: string]: string }] | string // console.log(Sentry)
type PathFn = (path: Path) => { data: object }

describe('GitHub Project Component', () => {
  test('renders correctly', async () => {
    let tree: ReactTestRenderer

    jest.unstable_mockModule('@octokit/core', () => ({
      Octokit: jest.fn().mockImplementation(() => ({
        request: jest.fn<PathFn>().mockImplementation((path: Path) => {
          let pathString: string
          if (Array.isArray(path)) {
            pathString = path[0]
          } else {
            pathString = path
          }

          switch (pathString) {
            case 'GET /repos/{owner}/{repo}':
              return {
                data: {
                  html_url: 'github.com/kochie/test',
                  name: 'test',
                  description: "test's description",
                  owner: {
                    login: 'kochie',
                    avatar_url:
                      'https://avatars.githubusercontent.com/u/10809884.jpg',
                    html_url: 'https://github.com/kochie',
                  },
                  stargazers_count: 100,
                  open_issues_count: 10,
                },
              }
            case 'GET /repos/{owner}/{repo}/contributors':
              return { data: new Array(10) }
            case 'GET /repos/{owner}/{repo}/languages':
              return {
                data: { TypeScript: 0.1, Go: 0.9 },
              }
            default:
              throw new Error('Unknown path' + path)
          }
        }),
      })),
    }))

    const GithubProject = await import('../index')

    await act(async () => {
      tree = create(<GithubProject.default owner="kochie" repo="test" />)
    })

    // @ts-expect-error tree will be assigned
    expect(tree.toJSON()).toMatchSnapshot()
  })
})
