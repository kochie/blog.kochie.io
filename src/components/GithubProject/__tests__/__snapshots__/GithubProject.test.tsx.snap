// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GitHub Project Component renders correctly 1`] = `
<div
  className="w-full rounded bg-white dark:bg-gray-500 relative overflow-hidden"
>
  <div
    className="m-8"
  >
    <div
      className="flex justify-between"
    >
      <div
        className=""
      >
        <div
          className="flex text-4xl mb-4"
        >
          <a
            className="flex hover:underline"
            href="github.com/kochie/test"
          >
            <div>
              kochie
            </div>
            <div>
              /
            </div>
            <div
              className="font-extrabold"
            >
              test
            </div>
          </a>
        </div>
        <div
          className="mb-4"
        >
          test's description
        </div>
      </div>
      <div
        className="h-16 w-16 rounded-3xl overflow-hidden"
      >
        <a
          href="https://github.com/kochie"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          <img
            alt="avatar"
            className="hover:scale-110 transform-gpu ease-in-out duration-100"
            data-nimg="1"
            decoding="async"
            height={100}
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src="/_next/image?url=https%3A%2F%2Favatars.githubusercontent.com%2Fu%2F10809884.jpg&w=320&q=75"
            srcSet="/_next/image?url=https%3A%2F%2Favatars.githubusercontent.com%2Fu%2F10809884.jpg&w=320&q=75 1x"
            style={
              {
                "color": "transparent",
                "height": "auto",
                "maxWidth": "100%",
              }
            }
            width={100}
          />
        </a>
      </div>
    </div>
    <div
      className="flex justify-between my-4"
    >
      <div
        className="flex gap-8"
      >
        <div
          className="flex group"
        >
          <div
            className=""
          >
            <div
              className="group-hover:scale-125 transform-gpu fa-stack ease-in-out duration-100"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-user-group fa-stack-1x group-hover:animate-wiggle"
                data-icon="user-group"
                data-prefix="far"
                focusable="false"
                role="img"
                style={{}}
                viewBox="0 0 640 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1.5 458.7c6-45.5 29.2-85.6 62.8-113.6C95.3 319.4 135 304 178.3 304H224h45.7c43.3 0 83 15.4 113.9 41.1c33.6 28 56.8 68.1 62.8 113.6c1 7.7 1.5 15.6 1.5 23.6c0 16.4-13.3 29.7-29.7 29.7H398.7 49.3 29.7C13.3 512 0 498.7 0 482.3c0-8 .5-15.9 1.5-23.6zM96 128a128 128 0 1 1 256 0A128 128 0 1 1 96 128zm208 0a80 80 0 1 0 -160 0 80 80 0 1 0 160 0zM178.3 352c-47.8 0-89.5 25.7-112.2 64c-8.6 14.4-14.4 30.7-16.8 48H98 350h48.8c-.4-2.7-.8-5.3-1.4-8c-2.9-14.3-8.2-27.8-15.5-40c-22.7-38.3-64.4-64-112.2-64H178.3zm431 160H472.4c4.9-8.8 7.6-18.9 7.6-29.7c0-65-29.5-123.2-75.9-161.8c4.3-.4 8.7-.5 13.2-.5h61.4C567.8 320 640 392.2 640 481.3c0 17-13.8 30.7-30.7 30.7zM432 256c-31 0-59-12.6-79.3-32.9C372.4 196.5 384 163.6 384 128c0-26.8-6.6-52.1-18.3-74.3C384.3 40.1 407.2 32 432 32c61.9 0 112 50.1 112 112s-50.1 112-112 112z"
                  fill="currentColor"
                  style={{}}
                />
              </svg>
            </div>
          </div>
          <div>
            <div
              className=""
            >
              10
            </div>
            <div
              className="text-sm"
            >
              Contributors
            </div>
          </div>
        </div>
        <div
          className="flex group"
        >
          <div
            className=""
          >
            <div
              className="group-hover:scale-125 transform-gpu fa-stack ease-in-out duration-100"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-circle-dot fa-stack-1x group-hover:animate-wiggle"
                data-icon="circle-dot"
                data-prefix="far"
                focusable="false"
                role="img"
                style={{}}
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M464 256A208 208 0 1 0 48 256a208 208 0 1 0 416 0zM0 256a256 256 0 1 1 512 0A256 256 0 1 1 0 256zm256-96a96 96 0 1 1 0 192 96 96 0 1 1 0-192z"
                  fill="currentColor"
                  style={{}}
                />
              </svg>
            </div>
          </div>
          <div>
            <div
              className=""
            >
              10
            </div>
            <div
              className="text-sm"
            >
              Issues
            </div>
          </div>
        </div>
        <div
          className="flex group"
        >
          <div
            className=""
          >
            <div
              className="group-hover:scale-125 transform-gpu fa-stack ease-in-out duration-100"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-star fa-stack-1x group-hover:animate-wiggle"
                data-icon="star"
                data-prefix="far"
                focusable="false"
                role="img"
                style={{}}
                viewBox="0 0 576 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M287.9 0c9.2 0 17.6 5.2 21.6 13.5l68.6 141.3 153.2 22.6c9 1.3 16.5 7.6 19.3 16.3s.5 18.1-5.9 24.5L433.6 328.4l26.2 155.6c1.5 9-2.2 18.1-9.6 23.5s-17.3 6-25.3 1.7l-137-73.2L151 509.1c-8.1 4.3-17.9 3.7-25.3-1.7s-11.2-14.5-9.7-23.5l26.2-155.6L31.1 218.2c-6.5-6.4-8.7-15.9-5.9-24.5s10.3-14.9 19.3-16.3l153.2-22.6L266.3 13.5C270.4 5.2 278.7 0 287.9 0zm0 79L235.4 187.2c-3.5 7.1-10.2 12.1-18.1 13.3L99 217.9 184.9 303c5.5 5.5 8.1 13.3 6.8 21L171.4 443.7l105.2-56.2c7.1-3.8 15.6-3.8 22.6 0l105.2 56.2L384.2 324.1c-1.3-7.7 1.2-15.5 6.8-21l85.9-85.1L358.6 200.5c-7.8-1.2-14.6-6.1-18.1-13.3L287.9 79z"
                  fill="currentColor"
                  style={{}}
                />
              </svg>
            </div>
          </div>
          <div>
            <div
              className=""
            >
              100
            </div>
            <div
              className="text-sm"
            >
              Stars
            </div>
          </div>
        </div>
      </div>
      <div
        className="my-auto"
      >
        <a
          aria-label="github"
          href="https://github.com"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-github fa-2x hover:scale-110 transform-gpu ease-in-out duration-100"
            data-icon="github"
            data-prefix="fab"
            focusable="false"
            role="img"
            style={{}}
            viewBox="0 0 496 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
              fill="currentColor"
              style={{}}
            />
          </svg>
        </a>
      </div>
    </div>
  </div>
  <div
    className="absolute bottom-0 w-full"
  >
    <span
      className="h-4 flex"
    >
      <span
        style={
          {
            "backgroundColor": "#2b7489",
            "width": "10%",
          }
        }
        title="TypeScript - 10.00%"
      />
      <span
        style={
          {
            "backgroundColor": "#00ADD8",
            "width": "90%",
          }
        }
        title="Go - 90.00%"
      />
    </span>
  </div>
</div>
`;
